### PR TITLE
Add HME support for ICLLP AVC VDEnc (refactored)

### PR DESCRIPTION
**Pull request by dvrogozh**
That's refactored version of https://github.com/intel/media-driver/pull/463.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -66,6 +66,16 @@ if (BUILD_KERNELS)
         message(FATAL_ERROR "Failed to find IGA")
     endif()
 
+    find_program(CMC cmc)
+    if (NOT CMC)
+        message(FATAL_ERROR "Failed to find CMC")
+    endif()
+
+    find_program(PYTHON python)
```
Comment (dvrogozh) [inline CMakeLists.txt]: I still worry about python version which we use...

File: media_driver/agnostic/media_srcs.cmake [hunk 2]
```diff
@@ -85,8 +88,103 @@ if (BUILD_KERNELS)
...
+
+        add_custom_command(
+            OUTPUT ${out_dir}
+            COMMAND ${CMAKE_COMMAND} -E make_directory ${out_dir}
+            COMMENT "Create codec kernels output directory: ${out_dir}")
+
+        # Compiling all the given sources
+        set(dats "")
+        if(genx STREQUAL "11")
+          set(genx "11lp")
```
Comment (dvrogozh) [inline media_driver/agnostic/media_srcs.cmake]: @cchunbo : I am **forced** to do this to match the generated kernel with your pregenerated one! As soon as I program honest -Qxcm_jit_target=gen11, resulted kernel binaries don't match. What's going on and for which platform we really generate the kernel?

File: media_driver/agnostic/media_srcs.cmake [hunk 3]
```diff
@@ -85,8 +88,103 @@ if (BUILD_KERNELS)
...
+            COMMAND KernelBinToSource -i ${krn}.krn -o ${krn_dir}/ -v ${krn} -index ${index} -t ${IDR_CODEC_TOTAL_NUM_KERNELS}
+            COMMENT "Generate source file from krn")
+    endfunction()
+
+    # List of kernel sources to build.
+    # NOTE: Order is important!! It should match the order in which sub-kernels are described
+    # in the corresponding strcuture in the driver. For example, HME kernel should
+    # match HmeDsScoreboardKernelHeaderG11
+    list(APPEND HME_KRN_SOURCES
+        ${CMAKE_CURRENT_LIST_DIR}/gen11/codec/kernel_free/Source/downscale_kernel_genx.cpp
```
Comment (dvrogozh) [inline media_driver/agnostic/media_srcs.cmake]: I don't see that IDR_CODEC_HME_DS_SCOREBOARD_KERNEL kernel is used anywhere else except gen11, so I think we can left these sources here for now.

File: media_driver/agnostic/gen11/codec/kernel_free/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+4. Build igc following the steps in readme @ https://github.com/intel/intel-graphics-compiler
+
+### Step2: Building Kernels
+
+1. Set envrionment variale (Set compiler path)
+
+   export PATH=$PATH:<cm_compiler_path>/include:<cm_compiler_path>/bin
+
+2. run build.py to generate kernel binary
```
Comment (yangnin) [inline media_driver/agnostic/gen11/codec/kernel_free/README.md]: Do we still use build.py in our patch? We already have merge.py.

Comment (dvrogozh) [reply media_driver/agnostic/gen11/codec/kernel_free/README.md]: we don't use build.py in cmake integration, but I did not want to rewrite this README.md: merge.py is not self-contained, we can't just replace build.py with merge.py. So, I decided to left build.py.

We can consider drop this README.md at all...

**Comment by dvrogozh** [comment]
Rebased.

**Pull request merged**